handler/src/main/java/io/netty/handler/ssl/PemPrivateKey.java

```diff
@@ -0,0 +1,186 @@
+        }
+
+        ByteBuf encoded = Unpooled.wrappedBuffer(key.getEncoded());
+        try {
+            ByteBuf pem = Unpooled.buffer(encoded.readableBytes());
+            pem.writeBytes(BEGIN_PRIVATE_KEY);
+            ByteBuf base64 = Base64.encode(encoded, true);
+            try {
+                pem.writeBytes(base64);
+            } finally {
+                SslUtils.zerooutAndRelease(base64);
+            }
+            pem.writeBytes(END_PRIVATE_KEY);
+            return valueOf(pem);
+        } finally {
+            SslUtils.zerooutAndRelease(encoded);
+        }
+    }
+
+    /**
+     * Creates a {@link PemPrivateKey} from raw {@code byte[]}.
+     *
+     * ATTENTION: It's assumed that the given argument is a PEM/PKCS#8 encoded value.
+     * No input validation is performed to validate it.
+     */
+    public static PemPrivateKey valueOf(byte[] key) {
+        return valueOf(Unpooled.wrappedBuffer(key));
+    }
+
+    /**
+     * Creates a {@link PemPrivateKey} from raw {@code ByteBuf}.
+     *
+     * ATTENTION: It's assumed that the given argument is a PEM/PKCS#8 encoded value.
+     * No input validation is performed to validate it.
+     */
+    public static PemPrivateKey valueOf(ByteBuf key) {
+        return new PemPrivateKey(key);
+    }
+
+    private final ByteBuf content;
+
+    private boolean destroyed;
+
+    private PemPrivateKey(ByteBuf content) {
+        this.content = content;
+    }
+
+    @Override
+    public ByteBuf content() {
+        int count = refCnt();
+        if (count <= 0) {
+            throw new IllegalReferenceCountException(count);
+        }
+
+        if (destroyed) {
+            throw new IllegalStateException();
+        }
+
+        return content;
+    }
+
+    @Override
+    public PemPrivateKey copy() {
+        return replace(content.copy());
+    }
+
+    @Override
+    public PemPrivateKey duplicate() {
+        return replace(content.duplicate());
+    }
+
+    @Override
+    public PemPrivateKey retainedDuplicate() {
+        return replace(content.retainedDuplicate());
+    }
+
+    @Override
+    public PemPrivateKey replace(ByteBuf content) {
+        return new PemPrivateKey(content);
+    }
+
+    @Override
+    public PemPrivateKey touch() {
+        return touch(null);
+    }
+
+    @Override
+    public PemPrivateKey touch(Object hint) {
+        content.touch(hint);
+        return this;
+    }
+
+    @Override
+    public PemPrivateKey retain() {
+        return (PemPrivateKey) super.retain();
+    }
+
+    @Override
+    public PemPrivateKey retain(int increment) {
+        return (PemPrivateKey) super.retain(increment);
+    }
+
+    @Override
+    protected void deallocate() {
+        destroy();
+        content.release();
+    }
+
+    @Override
+    public byte[] getEncoded() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public String getAlgorithm() {
+        throw new UnsupportedOperationException();
```

Comment:
can/should we support this based upon https://docs.oracle.com/javase/7/docs/api/java/security/Key.html#getAlgorithm() ?